packages/fetchai/connections/p2p_libp2p/check_dependencies.py

```diff
@@ -19,19 +19,130 @@
+    return next(islice(iterable, n, None), default)
+
+
+def get_version(*args: int) -> VERSION:
+    """
+    Get the version from a list of arguments.
+
+    Set to '0' if there are not enough arguments.
+    """
+    major = nth(args, 0, 0)
+    minor = nth(args, 1, 0)
+    patch = nth(args, 2, 0)
+    return major, minor, patch
+
+
+def version_to_string(version: VERSION) -> str:
+    """
+    Transform version to string.
+
+    :param version: the version.
+    :return: the string representation.
+    """
+    return ".".join(map(str, version))
+
+
+def print_ok_message(
+    binary_name: str, actual_version: VERSION, version_lower_bound: VERSION
+):
+    """
+    Print OK message.
+
+    :param binary_name: the binary binary_name.
+    :param actual_version: the actual version.
+    :param version_lower_bound: the version lower bound.
+    :return: None
+    """
+    print(
+        f"check '{binary_name}'>={version_to_string(version_lower_bound)}, found {version_to_string(actual_version)}"
+    )
+
+
+def check_binary(
+    binary_name: str,
+    args: List[str],
+    version_regex: Pattern,
+    version_lower_bound: VERSION,
+):
+    """
+    Check a binary is accessible from the terminal.
+
+    It breaks down in:
+    1) check if the binary is reachable from the system path;
+    2) check that the version number is higher or equal than the minimum required version.
+
+    :param binary_name: the name of the binary.
+    :param args: the arguments to provide to the binary to retrieve the version.
+    :param version_regex: the regex used to extract the version from the output.
+    :param version_lower_bound: the minimum required version.
+
+    :return: None
+    """
+    path = shutil.which(binary_name)
     if not path:
-        raise Exception(ERROR_MESSAGE_TEMPLATE.format(command=name))
+        raise AEAException(
+            ERROR_MESSAGE_TEMPLATE_BINARY_NOT_FOUND.format(command=binary_name)
+        )
+
+    version_getter_command = [binary_name, *args]
+    stdout = subprocess.check_output(version_getter_command).decode("utf-8")  # nosec
+    version_match = version_regex.search(stdout)
+    if version_match is None:
+        print(
+            f"Warning: cannot parse '{binary_name}' version from command: {version_getter_command}."
+        )
+        return
+    actual_version: VERSION = get_version(*map(int, version_match.groups(default="0")))
+    if actual_version < version_lower_bound:
+        raise AEAException(
+            ERROR_MESSAGE_TEMPLATE_VERSION_TOO_LOW.format(
+                command=binary_name,
+                lower_bound=version_to_string(version_lower_bound),
+                actual_version=version_to_string(actual_version),
+            )
+        )
+
+    print_ok_message(binary_name, actual_version, version_lower_bound)
+
+
+def main():
+    """The main entrypoint of the script."""
+    check_binary(
+        "go",
+        ["version"],
+        re.compile(r"go version go([0-9]+)\.([0-9]+) "),
```

Comment:
`go` doesn't provide a patch number; we default it to `0`, see how we use the `nth` function.